docs/markdown/Builtin-options.md

```diff
@@ -281,3 +282,14 @@ relative to the installation `prefix`, which will often result in installed pyth
 modules to not be found by the interpreter unless `prefix` is `/usr` on Linux,
 or for example `C:\Python39` on Windows. These options can be absolute paths
 outside of `prefix`.
+
+*Since 0.61.0* The `python.install_env` option is used to detect the correct
+installation path. Setting to `system` will avoid making the paths relative to
+`prefix` and instead use the global site-packages of the selected python
+interpreter directly, even if it is a venv. Setting to `venv` will instead use
+the paths for the virtualenv the python found installation comes from (or fail
+if it is not a virtualenv).  Setting to `auto` will check if the found
+installation is a virtualenv, and use `venv` or `system` as appropriate (but
+never `prefix`).
```

Comment:
In the sense that it's invalid to manually define both options at the same time, since they conceptually conflict?

The "correct" way to do that would be to check whether `python.install_env` has been set, rather than checking if its value is `prefix`, but of course we can't actually do that, right? We don't know whether an option has been defined on the command line. :/

Reply:
We have `interpreter.user_defined_options` for that.

Reply:
... I did not realize that was a thing. That being said, it feels like a rather inelegant way of approaching the get_option problem, and also... do we want to add more poking at the interpreter from inside a module?

Reply:
Hopefully implemented.